### ISSUE TITLE
Problem renewing token

### ISSUE THREAD

**Issue by SaxPlayer** [XeroAPI/xero-php-oauth2#131]
**SDK you're using (please complete the following information):**
 - Tried last stable of v1 and have also tried v2.0.0
 - PHP 7.3

**Describe the bug**
Process stopping when requesting new token after expiry. See details below and in screenshot.

**To Reproduce**
Wait for token to expire after 30 minutes. Code to refresh token is called. Gets to line indicated in screenshot and execution of script stops. No error is thrown or shown.

**Expected behavior**
Give new access token in $newAccessToken variable. Note, if I fudge the code and request a new token before the 30 minute expiry then the code runs and does not stop. It only seems to fail completely once the 30 minutes is up.

**Screenshots**
<img width="683" alt="Screenshot 2020-06-11 at 14 50 55" src="https://user-images.githubusercontent.com/47667566/84392996-f6d7e200-abf2-11ea-9a0b-840407c3deee.png">

**Additional context**
 - Everything works as expected within the 30 minute window. Connecting to Xero initially works, retrieving token, refreshtoken and expiry works and calling the API also works as expected. Problems happen once the token expires and I need to fetch a new one.
 - I've tried with the last stable version of v1 and have also tried v2.0.0
 - I've tried removing my app in Xero and then re-authorising it from scratch. Same problem.
 - I've tried with and without the line: 'grant_type' => 'refresh_token' (see screenshot)

**Comment by wobinb** [comment]
Hi

Thanks for including the full detail.

Firstly what are the contents of $newAccessToken? This should reveal if the API call to refresh the token was successful.
 
Assuming that the token refresh is successful, then the error is occurring when saving the new tokens. The default storage.php saves them as session variables, the idea being that in production you would store them somewhere more permanent. Could it be that your session is also expiring at 30 minutes?

**Comment by SaxPlayer** [comment]
Thanks for your reply. 

$newAccessToken doesn't return anything. The code stops execution completely at that point with no error (or value) returned.

If I echo("foobar"); just before the $newAccessToken = $provider... line then I see "foobar" in the output and then nothing else. If I move that echo line to after the $newAccessToken = $provider code then I get no output at all. So it's not getting as far as the $storage code.

For info, I've updated the storage class to use a database and so sessions aren't used for this. However, just for completeness I confimed that the PHP session itself is still active. 

Within the 30 minutes everything works fine and even the $newAccessToken line will work and return a value if I force that part of the code to run even though the token hasn't really expired. After the token's real expiry time everything simply stops at the line indicated with no errors.

**Comment by SaxPlayer** [comment]
I've made progress on debugging this issue by drilling down into AbstractProvider.php in oauth2-client/src/Provider

It appears that I wasn't storing the new Xero refresh token the very first time my code requested it after expiry. This means that when the getAccessToken() code ran, there was an "invalid_grant" error. However, this error doesn't appear to be returned when the function checkResponse() is called from getParsedResponse()

I don't want to second guess what the code is doing here exactly and it could very well be the way I'm (not) checking for errors, however perhaps the template code needs to be expanded to trap this error correctly?

In summary, to reproduce this problem call getAccessToken() by passing an old (or presumably invalid) refresh token.

**Comment by SidneyAllen** [comment]
@SaxPlayer - this sounds like feedback you should provide to the maintainers of the oauth2-client. We include this library, but you are free to use any OAuth2 php library you wish.

https://github.com/thephpleague/oauth2-client

**Comment by vijayadhithyanmohan1701** [comment]
Hi @SidneyAllen and @SaxPlayer,

I am still facing this issue. I started to migrate today and I am facing the issue in refreshing tokens. I have raised the issue here 
[#232]. It would be great if you could offer me some suggestions.
